### PR TITLE
Wundows&Unicode paths fix, Related to radareorg/cutter#589

### PR DESCRIPTION
**Pull request by a1ext**
Related to radareorg/cutter#589
--

Allow windows users to open unicode-named files from command shell
`pe` bin plugin fixed: wrong assignment of a file name
UTF8 has made a default charset for radare2 on windows if unicode build is enabled
fixed heap corruption in SDB
fixed proper project saving on windows

### CODE REVIEW

File: libr/util/file.c [hunk 1]
```diff
@@ -26,6 +26,25 @@
...
+	if (!file || !pStat) {
+		return -1;
+	}
+#if __WINDOWS__
+	wchar_t *wfile = r_utf8_to_utf16 (file);
+	int ret = -1;
+	if (!wfile) {
+		return -1;
+	}
+	ret = _wstat (wfile, pStat);
```
Comment (radare) [inline libr/util/file.c]: `int ret` in here to avoid the unnecessary assignment above

File: shlr/sdb/src/disk.c [hunk 2]
```diff
@@ -12,49 +12,34 @@
 #if __SDB_WINDOWS__
```
Comment (radare) [inline shlr/sdb/src/disk.c]: please,  if you do changes. in the sdb directory , make first a separate PR in the sdb repository

File: shlr/sdb/src/disk.c [hunk 3]
```diff
@@ -12,49 +12,34 @@
...
-	return res_buf;
-}
+	wchar_t *rutf16 = NULL;
+	int wcsize;
 
-static wchar_t* r_str_mb_to_wc(const char *buf) {
-	if (!buf) {
-		return NULL;
+	if ((wcsize = MultiByteToWideChar (CP_UTF8, 0, cstring, len, NULL, 0))) {
+		wcsize += 1;
```
Comment (radare) [inline shlr/sdb/src/disk.c]: ++

File: shlr/sdb/src/disk.c (hunk 2 above)
Comment (a1ext) [reply shlr/sdb/src/disk.c]: should I remove these changes from this PR?

Comment (a1ext) [reply shlr/sdb/src/disk.c]: done https://github.com/radare/sdb/pull/170

**Comment by radare** [comment]
Please solve conflicts (i have updated sdb in master)

**Comment by a1ext** [comment]
done

**Comment by radare** [comment]
thanks!
